packages/search-ui-app-search-connector/search-ui-app-search-connector.d.ts

```diff
@@ -0,0 +1,65 @@
+declare module "@elastic/search-ui-app-search-connector" {
+  class AppSearchAPIConnector {
+    /**
+     * @typedef {Object.<string, *>} KeyValuePairs
+     */
+    /**
+     * @callback next
+     * @param {KeyValuePairs} updatedQueryOptions The options to send to the API
+     * @returns {Object} The response
+     */
+    /**
+     * @callback hook
+     * @param {KeyValuePairs} queryOptions The options that are about to be sent to the API
+     * @param {next} next The options that are about to be sent to the API
+     */
+    /**
+     * @param {Object} options Credential found in your App Search Dashboard
+     * @param {string} options.searchKey Credential found in your App Search Dashboard
+     * @param {string} options.engineName Engine to query, found in your App Search Dashboard
+     * @param {string=} options.hostIdentifier Credential found in your App Search Dashboard. Required only if not using
+     * the endpointBase option.
+     * @param {hook=} options.beforeSearchCall A hook to amend request or response to API for "onSearch" event.
+     * @param {hook=} options.beforeAutocompleteResultsCall A hook to amend request or response to API for a "results"
+     * query on an "onAutocomplete" event.
+     * @param {hook=} options.beforeAutocompleteSuggestionsCall A hook to amend request or response to API for a "suggestions"
+     * query on an "onAutocomplete" event.
+     * @param {string=} options.endpointBase="" Overrides the base of the Swiftype API endpoint completely. Useful
+     * for non-SaaS App Search deployments. For example, Elastic Cloud, Self-Managed, or proxying SaaS.
+     */
+    constructor({
+      searchKey,
+      engineName,
+      hostIdentifier,
+      beforeSearchCall,
+      beforeAutocompleteResultsCall,
+      beforeAutocompleteSuggestionsCall,
+      endpointBase
+    }: {
+      searchKey: string;
+      engineName: string;
+      hostIdentifier?: string;
+      beforeSearchCall?: (
+        queryOptions: {
```

Comment:
`{ [x: string]: any }` is equivalent to `object`. Usually the latter is preferred

Reply:
@alexanderatallah That's fair, I think I will make that change. So type `Object` in uppercase, or `object` in lower case?

Reply:
@JasonStoltz `object` in lower case is better. According to https://github.com/DefinitelyTyped/DefinitelyTyped:
> Using the types Function and Object is almost never a good idea. In 99% of cases it's possible to specify a more specific type. Examples are (x: number) => number for functions and { x: number, y: number } for objects. If there is no certainty at all about the type, any is the right choice, not Object. If the only known fact about the type is that it's some object, use the type object, not Object or { [key: string]: any }.